### PR TITLE
Refactor fountain painting

### PR DESCRIPTION
**Pull request by duncanspumpkin**
little refactor

### CODE REVIEW

File: src/openrct2/paint/sprite/Paint.Misc.cpp [hunk 1]
```diff
@@ -108,63 +108,26 @@ void misc_paint(paint_session* session, const rct_sprite* misc, int32_t imageDir
...
-                switch (ebx)
-                {
-                    case 0:
-                        sub_98197C(session, imageId, 0, 0, 32, 1, 3, height, -32, -3, height);
-                        break;
-
-                    case 1:
-                        sub_98197C(session, imageId, 0, 0, 1, 32, 3, height, -3, 0, height);
-                        break;
+            constexpr std::array<CoordsXY, 2> antiClockWiseBoundingBoxes = { CoordsXY{ -32, -3 }, CoordsXY{ 0, -3 } };
```
Comment (tupaschoal) [inline src/openrct2/paint/sprite/Paint.Misc.cpp:125]:
```suggestion
            constexpr std::array<CoordsXY, 2> antiClockWiseBoundingBoxes = { CoordsXY{ -COORDS_XY_STEP, -3 }, CoordsXY{ 0, -3 } };
```

File: src/openrct2/paint/sprite/Paint.Misc.cpp [hunk 2]
```diff
@@ -108,63 +108,26 @@ void misc_paint(paint_session* session, const rct_sprite* misc, int32_t imageDir
...
+            constexpr std::array<CoordsXY, 2> clockWiseBoundingBoxes = { CoordsXY{ -32, 3 }, CoordsXY{ 0, 3 } };
```
Comment (tupaschoal) [inline src/openrct2/paint/sprite/Paint.Misc.cpp:126]:
```suggestion
            constexpr std::array<CoordsXY, 2> clockWiseBoundingBoxes = { CoordsXY{ -COORDS_XY_STEP, 3 }, CoordsXY{ 0, 3 } };
```

File: src/openrct2/world/Fountain.h [hunk 3]
```diff
@@ -41,3 +41,13 @@ enum
     JUMPING_FOUNTAIN_TYPE_WATER,
     JUMPING_FOUNTAIN_TYPE_SNOW
 };
+
+namespace FOUNTAIN_FLAG
```
Comment (tupaschoal) [inline src/openrct2/world/Fountain.h:45]: Might be a good time to turn it into an `enum`?

Comment (duncanspumpkin) [reply src/openrct2/world/Fountain.h:45]: Sadly not possible to be done with flags. The preferred style is a namespace of constexprs

**Comment by duncanspumpkin** [comment]
Does COORDSXY_STEP make sense in these situations?

**Comment by tupaschoal** [comment]
I can't tell, it seems like it isn't, so I wouldn't mind if you skipped these changes.

**Comment by duncanspumpkin** [comment]
In the end decided to make those changes. I don't think they quite make sense as its the size of the fountain and it just happens to be the size of a box.

File: src/openrct2/paint/sprite/Paint.Misc.cpp [hunk 4]
```diff
@@ -108,63 +108,26 @@ void misc_paint(paint_session* session, const rct_sprite* misc, int32_t imageDir
...
-                switch (ebx)
-                {
-                    case 0:
-                        sub_98197C(session, imageId, 0, 0, 32, 1, 3, height, -32, -3, height);
-                        break;
-
-                    case 1:
-                        sub_98197C(session, imageId, 0, 0, 1, 32, 3, height, -3, 0, height);
-                        break;
+            constexpr std::array<CoordsXY, 2> antiClockWiseBoundingBoxes = { CoordsXY{ -COORDS_XY_STEP, -3 }, CoordsXY{ 0, -3 } };
```
Comment (tupaschoal) [inline src/openrct2/paint/sprite/Paint.Misc.cpp:125]:
```suggestion
            constexpr std::array<CoordsXY, 2> antiClockWiseBoundingBoxes = { CoordsXY{ -COORDS_XY_STEP, -3 },
                                                                             CoordsXY{ 0, -3 } };
```

**Pull request merged**